### ISSUE TITLE
[CLOSED] Kernel should allow hash/version from out-of-band DApp storage systems

### ISSUE THREAD

**Issue by aionbot** [aionnetwork/AVM#235]
**Issue created by [yulongaion](https://github.com/yulongaion)** (on Tuesday Sep 04, 2018 at 20:24 GMT)

To replace the current Aion `ContractDetailStorage`

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Wednesday Sep 12, 2018 at 16:23 GMT)

I suspect that this is what is referenced, here:  https://github.com/aionnetwork/aion/blob/dev-debug/modAionBase/src/org/aion/base/db/IContractDetails.java

In either case, we will base this on the reception of the #246 proof-of-concept, so this is blocked on that.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Friday Oct 26, 2018 at 20:23 GMT)

Since #265 talks about building a specific back-end, and we already have the `IObjectGraphStore` interface, this item should be formalized into a change in how the kernel stores information regarding the DApp application storage so that it can reference other implementations.  Specifically, in our case, this means that we can do the 2 following things:

1. Pass in an arbitrary "storage hash" when updating an account state
1. Disable the existing Merkle tree hash computation in the key-value store on a per-DApp basis (since we might want to use that to support our graph)

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Monday Nov 19, 2018 at 14:38 GMT)

We will need this well before TestNet but it doesn't currently gate Beta0 so it can be deferred to Beta1.

**Comment by aionbot** [comment]
**Comment by [nancyaion](https://github.com/nancyaion)** (on Friday Nov 23, 2018 at 22:19 GMT)

kernel commit 408e6079a142b2108773809eb4fd664a8d4583c0

Add the storage root hash to the block summary and the transaction receipt

This commit also contains the unit test "HashCodeIntegrationTest" on the kernel side.

**Comment by aionbot** [comment]
**Comment by [nancyaion](https://github.com/nancyaion)** (on Thursday Nov 29, 2018 at 22:10 GMT)

kernel commit c2dfcbc4e2e33dfcb89a5c91ffe5f1d39ed7d150

Correct the logging info

**Comment by aionbot** [comment]
**Comment by [nancyaion](https://github.com/nancyaion)** (on Friday Nov 30, 2018 at 19:23 GMT)

Also verified via the Beta build that the storage hash code is present in the kernel blocks / receipts.
